processor/metricstransformprocessor/README.md

```diff
@@ -0,0 +1,118 @@
+action: {update, insert}
+
+# new_name is used to rename metrics (e.g. rename cpu/usage to cpu/usage_time)
+# if action is insert, new_name is required
+new_name: <new_metric_name_inserted>
+
+# operations contain a list of operations that will be performed on the selected 
+# metrics. Each operation block is a key-value pair, where the key can be any 
+# arbitrary string set by the users for readability, and the value is a struct with # fields required for operations. The action field is important for the processor 
+# to identify exactly which operation to perform 
+operations:
+
+  # update_label action can be used to update the name of a label or the values           
+  # of this label (e.g. rename label `cpu` to `core`)
+  -action: update_label
+   label: <current_label1>
+   new_label: <new_label>
+   value_actions:
+     -value: <current_label_value>
+      new_value: <new_label_value>
+
+  # aggregate_labels action aggregates metrics across labels (e.g. only want  
+  # the label `usage`, but don’t care about the labels `core`, and `cpu`)
+  -action: aggregate_labels
+   # label_set contains a list of labels that will remain after the aggregation.    
+   # The excluded labels will be aggregated by the way specified by  
+   # aggregation_type.
+   label_set: [labels...]
+   aggregation_type: {sum, average, max}
+
+  # aggregate_label_values action aggregates labels across label values (e.g. want  
+  # memory{slab}, but don’t care about memory{slab_reclaimable} &   
+  # memory{slab_unreclaimable})
+  -action: aggregate_label_values
+   label: <label>
+   # aggregated_values contains a list of label values that will be aggregated by  
+   # the way specified by aggregation_type into new_value. The excluded label  
+   # values will remain.
+   aggregated_values: [values...]
+   new_value: <new_value> 
+   aggregation_type: {sum, average, max}
+```
+
+## Examples
+
+### Insert New Metric
+```yaml
+# create host/cpu/usage_time from host/cpu/usage
+name: host/cpu/usage
+action: insert
+new_name: host/cpu/utilization
+operations:
+  ...
+```
+
+### Rename Labels
+```yaml
+# rename the label cpu to core
+operations:
+  - action: update_label
+    label: cpu
+    new_label: core
+```
+
+### Aggregate Labels
+```yaml
+# aggregate away everything but `state` using summation
+...
+operations:
+  -action: aggregate_labels
+   label_set: [ state ]
+   aggregation_type: sum
+```
+
+### Aggregate Label Values
+```yaml
+# combine slab_reclaimable & slab_unreclaimable by summation
+...
+operations:
+  -action: aggregate_label_values
+   label: state
+   aggregated_values: [ slab_reclaimable, slab_unreclaimable ]
+   new_value: slab 
+   aggregation_type: sum
+```
+
+## Possible Extensions
+- Supporting custom types of aggregation (e.g. defined by a formula)
+- Support aggregations of non-simple metric types (distributions, etc)
+- Support aggregation over time
+- Utilizing regex to select metrics
+
+## Thanks to
```

Comment:
Though I appreciate this section (and attribution is good general principle) what would you think about removing this from the public readme since once this is public the attribution is usually done via the commit history?